src/utxo/rpc.ts

```diff
@@ -0,0 +1,121 @@
+import { hex } from "@scure/base"
+import type { BitcoinMerkleProofResponse } from "../types/bitcoin.js"
+import { buildBitcoinMerkleProof, type UtxoDepositProof } from "./index.js"
+
+export interface UtxoRpcConfig {
+  url: string
+  headers?: Record<string, string>
+}
+
+type JsonRpcResponse<T> = {
+  jsonrpc: "2.0"
+  id: string | number | null
+  result?: T
+  error?: { code: number; message: string; data?: unknown }
+}
+
+interface RawTransactionResult {
+  blockhash?: string
+  hex: string
+  height?: number
+  vout?: Array<{ n: number; value: number }>
+}
+
+interface BlockResult {
+  tx: string[]
+  height?: number
+}
+
+export class UtxoRpcClient {
+  constructor(private readonly config: UtxoRpcConfig) {}
+
+  async call<T>(method: string, params: unknown[] = []): Promise<T> {
+    const response = await fetch(this.config.url, {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        ...(this.config.headers ?? {}),
+      },
+      body: JSON.stringify({ jsonrpc: "2.0", id: "1", method, params }),
+    })
+
+    if (!response.ok) {
+      throw new Error(`UTXO RPC request failed: ${response.status} ${response.statusText}`)
+    }
+
+    const body = (await response.json()) as JsonRpcResponse<T>
+    if (body.error) {
+      throw new Error(`UTXO RPC error: ${body.error.message}`)
+    }
+    if (body.result === undefined || body.result === null) {
+      throw new Error("UTXO RPC: missing result in response")
+    }
+    return body.result
+  }
+
+  async getTransaction(txHash: string): Promise<RawTransactionResult> {
+    return await this.call<RawTransactionResult>("getrawtransaction", [txHash, true])
```

Comment:
For zcash you have to send an integer (1) instead of a boolean

Reply:
But now it won't work for Bitcoin I think (but better check)? In rust SDK we chose to store chain in the client struct